backend/src/api/mutation.rs

```diff
@@ -275,6 +276,94 @@ impl Mutation {
         Ok(CreateRealmLineageOutcome { num_created })
     }
 
+    async fn prepare_series(series: NewSeries, context: &Context) -> ApiResult<PreparedSeries> {
+        if context.auth != AuthContext::TrustedExternal {
+            return Err(not_authorized!("only trusted external applications can use this mutation"));
+        }
+
+        let series = Series::create(series, context).await?;
+
+        Ok(PreparedSeries { id: series.id() })
+    }
+
+    async fn add_series_mount_point(
+        series_oc_id: String,
+        target_path: String,
+        context: &Context,
+    ) -> ApiResult<Realm> {
+        if context.auth != AuthContext::TrustedExternal {
+            return Err(not_authorized!("only trusted external applications can use this mutation"));
+        }
+
+        let series = Series::load_by_opencast_id(series_oc_id, context)
+            .await?
+            .ok_or_else(|| invalid_input!("`seriesId` does not refer to a valid series"))?;
+
+        let target_realm = Realm::load_by_path(target_path, context)
+            .await?
+            .ok_or_else(|| invalid_input!("`targetRealmPath` does not refer to a valid realm"))?;
+
+        BlockValue::add_series(
+            Id::realm(target_realm.key),
+            0,
+            NewSeriesBlock {
+                series: series.id(),
+                show_title: false,
+                show_metadata: true,
+                order: VideoListOrder::NewToOld,
+                layout: VideoListLayout::Gallery,
+            },
+            context,
+        ).await?;
+
+        let block = &BlockValue::load_for_realm(target_realm.key, context).await?[0];
+
+        Realm::rename(
+            target_realm.id(),
+            UpdatedRealmName::from_block(block.id()),
+            context,
+        ).await
+    }
+
+    async fn remove_series_mount_point(
+        path: String,
+        context: &Context,
+    ) -> ApiResult<RemoveMountedSeriesOutcome> {
+        if context.auth != AuthContext::TrustedExternal {
+            return Err(not_authorized!("only trusted external applications can use this mutation"));
+        }
+
+        let old_realm = Realm::load_by_path(path, context)
+            .await?
+            .ok_or_else(|| invalid_input!("`currentRealmPath` does not refer to a valid realm"))?;
+
+        let blocks = BlockValue::load_for_realm(old_realm.key, context).await?;
+
+        if blocks.len() > 1 {
+            return Err(invalid_input!("series can only be moved if its current realm has no other blocks"));
+        }
+
+        let mut removed_realm = None;
+        let mut removed_block = None;
+
+        if old_realm.children(context).await?.len() == 0 {
+            // The realm has no children, so it can be removed.
+            removed_realm = Some(Realm::remove(old_realm.id(), context).await?);
+        } else {
+            // At this point we can be certain that there is only one block, which is the series block.
```

Comment:
Actually no, these checks are not happening in this method, at this point we only know that there is 0 or 1 blocks in this realm and that the realm has some children. If the block has 0 blocks, the `blocks[0]` expression below panics. And the check whether the block is even mentioning the series is probably also useful? Also, the realm doesn't need to be renamed if it doesn't derive the name from the block.